spec/models/classification_spec.rb

```diff
@@ -61,14 +61,41 @@
       FactoryGirl.create(:classification_tag,      :name => "multi_entry_2", :parent => parent)
     end
 
-    it "#destroy" do
-      cat = Classification.find_by_name("test_category")
-      expect(cat).to_not be_nil
-      entries = cat.entries
-      expect(entries.length).to eq(2)
+    context "#destroy" do
+      it "a category deletes all entries" do
+        cat = Classification.find_by_name("test_category")
+        expect(cat).to_not be_nil
+        entries = cat.entries
+        expect(entries.length).to eq(2)
+
+        cat.destroy
+        entries.each { |e| expect(Classification.find_by(:id => e.id)).to be_nil }
+      end
+
+      it "a category deletes assignments referenced by its entries" do
+        cat = Classification.find_by_name("test_category")
+        assignment_tag = "/chargeback_rate/assigned_to/vm/tag/managed/test_category/test_entry"
+
+        Tag.create!(:name => assignment_tag)
+        expect(Tag.where(:name => assignment_tag).take).to_not be_nil
 
-      cat.destroy
-      entries.each { |e| expect(Classification.find_by(:id => e.id)).to be_nil }
+        cat.destroy
+
+        expect(Tag.where(:name => assignment_tag).take).to be_nil
+      end
+
+      it "an entry deletes assignments where its tag is referenced" do
+        cat = Classification.find_by_name("test_category")
+        ent = cat.entries.find { |e| e.name == "test_entry" }
+        assignment_tag = "/chargeback_rate/assigned_to/vm/tag/managed/test_category/test_entry"
+
+        Tag.create!(:name => assignment_tag)
+        expect(Tag.where(:name => assignment_tag).take).to_not be_nil
+
+        ent.destroy
+
+        expect(Tag.where(:name => assignment_tag).take).to be_nil
```

Comment:
Do we need a test case showing that a similar tag in a different category isn't removed?

For example, `"/chargeback_rate/assigned_to/vm/tag/managed/test_category2/test_entry"`

Reply:
The test I added to the `AssignmentMixin` tests that. Let me know if you think it still needs more. See https://github.com/ManageIQ/manageiq/pull/16039/files#diff-c96ad7d76c6072d9695e81a29f51e8b0R31

Reply:
Ah, I guess I was more concerned with destroy not deleting things that are "close" but not exactly the same tag.

Reply:
Ok, I'll add that test. Thanks!